driver-core/src/main/com/mongodb/client/model/expressions/ArrayExpression.java

```diff
@@ -58,67 +57,267 @@
+     * necessary, then the identify function {@code array.join(v -> v)} should
+     * be used.
+     *
+     * @param mapper the mapper function.
+     * @return the resulting value.
+     */
     StringExpression join(Function<? super T, StringExpression> mapper);
 
+    /**
+     * The array-concatenation of all the array values of {@code this} array,
+     * via the provided {@code mapper}. Returns the empty array if the array
+     * is empty.
+     *
+     * <p>The mapper may be used to transform the values of {@code this} array
+     * into {@linkplain ArrayExpression arrays}. If no transformation is
+     * necessary, then the identify function {@code array.concat(v -> v)} should
+     * be used.
+     *
+     * @param mapper the mapper function.
+     * @return the resulting value.
+     * @param <R> the type of the elements of the array.
+     */
     <R extends Expression> ArrayExpression<R> concat(Function<? super T, ? extends ArrayExpression<? extends R>> mapper);
 
+    /**
+     * The set union of all the array values of {@code this} array,
+     * via the provided {@code mapper}. Returns the empty array if the array
+     * is empty.
+     *
+     * <p>The mapper may be used to transform the values of {@code this} array
+     * into {@linkplain ArrayExpression arrays}. If no transformation is
+     * necessary, then the identify function {@code array.union(v -> v)} should
+     * be used.
+     *
+     * @param mapper the mapper function.
+     * @return the resulting value.
+     * @param <R> the type of the elements of the array.
+     */
     <R extends Expression> ArrayExpression<R> union(Function<? super T, ? extends ArrayExpression<? extends R>> mapper);
 
+    /**
+     * The {@linkplain MapExpression map} value corresponding to the
+     * {@linkplain EntryExpression entry} values of {@code this} array,
+     * via the provided {@code mapper}. Returns the empty array if the array
+     * is empty.
+     *
+     * <p>The mapper may be used to transform the values of {@code this} array
+     * into {@linkplain EntryExpression entries}. If no transformation is
+     * necessary, then the identify function {@code array.union(v -> v)} should
+     * be used.
+     *
+     * @see MapExpression#entrySet()
+     * @param mapper the mapper function.
+     * @return the resulting value.
+     * @param <R> the type of the resulting map's values.
+     */
     <R extends Expression> MapExpression<R> asMap(Function<? super T, ? extends EntryExpression<? extends R>> mapper);
 
     /**
-     * user asserts that i is in bounds for the array
+     * Returns the element at the provided index {@code i} for
+     * {@code this} array.
      *
-     * @param i
-     * @return
+     * <p>Warning: The use of this method is an assertion that
+     * the index {@code i} is in bounds for the array.
+     * If the index is out of bounds for this array, then
+     * the behaviour of the API is not defined.
```

Comment:
Let's replace "defined" with "specified" here and in other places to be consistent with `MqlUnchecked` and to avoid allusions to ["undefined behavor"](https://en.cppreference.com/w/c/language/behavior) in C, that puts no restrictions are put on the behavior of the program.

Reply:
4 instances in directory "is not defined" replaced.